src/components/producerItems/components/ProductEditModal.js

```diff
@@ -93,41 +99,96 @@ class ProductEditForm extends React.Component {
+
+            validateFields((error, fieldValues) => {
+                if (!error) {
+                    updateProduct({ ...productToEdit, ...fieldValues, image_id, image_url });
+                    return resetFields();
+                }
+                return error;
+            });
+        } else {
+            validateFields((error, fieldValues) => {
+                if (!error) {
+                    updateProduct({ ...productToEdit, ...fieldValues });
+                    return resetFields();
+                }
+                return error;
+            });
+        }
     }
+    
+      handleCancel = () => this.setState({ previewVisible: false })
+    
+      handlePreview = () => this.setState({ previewVisible: true });
+
+      handleRemove = () => this.setState({ fileDetails: [] })
+    
+      handleChange = info => {
+          if (info.file.status !== 'uploading') {
+              this.setState({ previewImage: 'info.file.response.data.image_url || info.thumbUrl' });
+          }
+          if (info.file.status === 'done') {
+              this.setState({ fileDetails: info.fileList });
+              message.success(`${info.file.name} file uploaded successfully`);
+          } else if (info.file.status === 'error') {
+              message.error(`${info.file.name} file upload failed.`);
+          }
+      }
+
+      static getDerivedStateFromProps(props) {
+          const { productToEdit, form: { getFieldDecorator } } = props;
+          const formItems = generateProductEditForm(getFieldDecorator, productToEdit);
+
+          return { formItems };
+      }
+
+      render() {
+          const { closeModal, isNewProduct, isOpen, isProductUpdating } = this.props;
+          const { formItems, previewVisible, previewImage, fileDetails } = this.state;
+          const title = isNewProduct ? CREATE_TITLE : EDIT_TITLE;
+
+          const uploadButton = (
+              <div>
+                  <Icon type="plus" />
+                  <div className="ant-upload-text">{"Add Image"}</div>
+              </div>
+          );
+
+          const props = {
+              action:"http://localhost:4000/api/createImage",
+              listType:"picture-card",
+              multiple:false,
+              name:"image",
+              onChange: this.handleChange,
+              onPreview: this.handlePreview,
+              onRemove: this.handleRemove,
+          };
+
+          return (
+              <Modal
+                  confirmLoading={isProductUpdating}
+                  visible={isOpen}
+                  title={title}
+                  okText={SAVE_PRODUCT_DETAILS}
+                  onCancel={closeModal}
+                  onOk={this.updateProductInfo}
+              >
+                  <Form>{formItems}</Form>
+                  <div className="clearfix">
+                      <Upload {...props}>
+                          {fileDetails.length >= 1 ? null : uploadButton}
```

Comment:
try this
```
{fileDetails.length >= 1 && uploadButton}
```

Reply:
I  want to show button only when the array is zero or one else hide button

Reply:
```
{fileDetails.length <= 1 && uploadButton}
``

Reply:
also make the compare value (in this case `1`) a variable given that it will change in the future. Store it in the constants file